ksql-engine/src/test/java/io/confluent/ksql/function/udaf/topk/DoubleTopkKudafTest.java

```diff
@@ -16,82 +16,85 @@
 
 package io.confluent.ksql.function.udaf.topk;
 
+import com.google.common.collect.ImmutableList;
+
 import io.confluent.ksql.function.KsqlAggregateFunction;
 import org.apache.kafka.connect.data.Schema;
 import org.junit.Before;
 import org.junit.Test;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.junit.Assert.assertThat;
 
 public class DoubleTopkKudafTest {
-  Object[] valueArray;
-  TopKAggregateFunctionFactory topKFactory;
-  List<Schema> argumentType;
+  private final List<Double> valuesArray = ImmutableList.of(10.0, 30.0, 45.0, 10.0, 50.0, 60.0, 20.0, 60.0, 80.0, 35.0,
+      25.0);
+  private TopKAggregateFunctionFactory topKFactory;
+  private List<Schema> argumentType;
 
   @Before
   public void setup() {
-    valueArray = new Double[]{10.0, 30.0, 45.0, 10.0, 50.0, 60.0, 20.0, 60.0, 80.0, 35.0, 25.0};
     topKFactory = new TopKAggregateFunctionFactory(3);
     argumentType = Collections.singletonList(Schema.FLOAT64_SCHEMA);
   }
 
   @Test
   public void shouldAggregateTopK() {
-    KsqlAggregateFunction<Object, Object[]> topkKudaf =
-            topKFactory.getProperAggregateFunction(argumentType);
-    Object[] window = new Double[]{null, null, null};
-    for (Object value : valueArray) {
+    KsqlAggregateFunction<Object, List<Double>> topkKudaf =
+        topKFactory.getProperAggregateFunction(argumentType);
```

Comment:
@dguy had the following comment on using getProperAggregateFunction:
Looks like we are missing some type information on AggregateFunctionFactory i.e., getProperAggregateFunction should return a KsqlAggregateFunction<V, A> then we get rid of all the warning noise in this class etc.

@dguy my question is: How would you add types to AggregateFunctionFactory? It's subclasses can have multiple types for the given method.

Reply:
Personally, I'm happy with this being left for another PR. This is not related to your change.